migration/api.py

```diff
@@ -44,22 +86,19 @@ def get_results_from_pages(
 
         while more_pages:
             logger.debug(f'Params: {extra_params}')
-            try:
-                resp = self.client.get(url=endpoint, params=extra_params, timeout=10)
-                resp.raise_for_status()
-            except httpx.HTTPError as exc:
-                logger.error(f"HTTP Exception for {exc.request.url} - {exc}")
-                raise exc
-            results += resp.json()
-            next_page_params = self.get_next_page_params(resp)
-            if next_page_params is None:
+            data = self.get(url=endpoint, params=extra_params)
+            results += data.data
+            if data.next_page_params is None:
                 more_pages = False
-            elif bail_after is not None and bail_after <= len(results):
+            elif limit is not None and limit <= len(results):
                 more_pages = False
             else:
-                extra_params.update(next_page_params)
+                extra_params.update(data.next_page_params)
                 page_num += 1
 
+        if limit is not None and len(results) > limit:
+            results = results[:limit]
```

Comment:
If `limit` is specified, I'm now actually slicing off any extra data, which I wasn't doing before.

Reply:
Sure, done. https://github.com/tl-its-umich-edu/tool-migration/pull/7/commits/48c65e27d544ef050b40d16ebe5925cbde2c77ac

Reply:
Thanks!